### PR TITLE
Jira skip marker

### PR DESCRIPTION
**Pull request by jabia99**
Added a new marker that lets you skip a test if a Jira ticket is not closed (can be more than one)

I also refactored the code to make it easier to read

### CODE REVIEW

File: test_suite/conftest.py [hunk 1]
```diff
@@ -2,85 +2,140 @@
...
+def __check_exclude_on_marker(exclude_on_marker, host_info):
+    if not exclude_on_marker:
+        return
+
+    exclude_on_marker_list = exclude_on_marker.args[0]
+
+    if host_info['distro'] in exclude_on_marker_list or host_info['distro_version'] in exclude_on_marker_list:
+        pytest.skip(host_info['skip_message'])
+
+    for item in exclude_on_marker_list:
```
Comment (narmaku) [inline test_suite/conftest.py:49]: As discussed in our call, let's try to simplify this for in both functions.

**Review by narmaku — CHANGES_REQUESTED** [review]

File: test_suite/conftest.py [hunk 2]
```diff
@@ -2,85 +2,140 @@
...
+    s = requests.Session()
+    retries = Retry(total=3, backoff_factor=3)
+    s.mount('http://', HTTPAdapter(max_retries=retries))
+
+    JIRA_PAT = os.getenv("JIRA_PAT")
+    if not JIRA_PAT:
+        exit("JIRA_PAT was not found")
+
+    headers = {'Authorization': f'Bearer {os.getenv("JIRA_PAT")}'}
+    endpoint_base = 'https://issues.redhat.com/rest/api/2/issue/'
```
[the remark this answers is not in the document]
Comment (jabia99) [reply test_suite/conftest.py:109]: issues.redhat.com is public. In fact, we are migrating from bugzilla to JIRA, customers will start using this page.

File: test_suite/conftest.py [hunk 3]
```diff
@@ -2,85 +2,140 @@
...
+            pytest.skip(f'Test skipped because Jira ticket {ticket_id} is not Closed yet')
+        else:
+            print(f'WARNING: Jira ticket {ticket_id} is already closed. Please remove it from the marker and put it in the docstring')
+
+
+@pytest.fixture(autouse=True, scope='function')
+def check_markers(host, request):
+    host_info = __get_host_info(host)
+
+    wait_marker = request.node.get_closest_marker('wait')
```
[the remark this answers is not in the document]
Comment (jabia99) [reply test_suite/conftest.py:129]: Definitely! thanks

File: test_suite/conftest.py [hunk 4]
```diff
@@ -2,85 +2,140 @@
...
+    if not jira_skip_marker:
         return
-    else:
-        pytest.skip(skip_message.format(distro_version=distro_version))
+
+    jira_skip_marker_list = jira_skip_marker.args[0]
+
+    s = requests.Session()
+    retries = Retry(total=3, backoff_factor=3)
+    s.mount('http://', HTTPAdapter(max_retries=retries))
```
[the remark this answers is not in the document]
Comment (jabia99) [reply test_suite/conftest.py:102]: On the multiple examples in the documentation they always use http as it is more general. It doesn't really matter as we are only using this api. But I'll use `https://issues.redhat.com` to check the server's certificate

File: test_suite/conftest.py [hunk 5]
```diff
@@ -2,85 +2,140 @@
...
+
+    headers = {'Authorization': f'Bearer {os.getenv("JIRA_PAT")}'}
+    endpoint_base = 'https://issues.redhat.com/rest/api/2/issue/'
+
+    for ticket_id in jira_skip_marker_list:
+        endpoint = endpoint_base + ticket_id
+        res = s.get(endpoint, headers=headers)
+
+        if res.status_code != 200:
+            exit(f'Could not check Jira ticket {ticket_id}. API request error code {res.status_code}')
```
[the remark this answers is not in the document]
Comment (jabia99) [reply test_suite/conftest.py]: Makes sense:)

**Pull request merged**